rskj-core/src/test/java/co/rsk/peg/BridgeSupportFlyoverTest.java

```diff
@@ -646,6 +646,139 @@ private BigInteger sendFundsToAnyAddress(
+        ReleaseTransactionSet releaseTransactionSet = new ReleaseTransactionSet(new HashSet<>());
+        when(provider.getReleaseTransactionSet()).thenReturn(releaseTransactionSet);
+
+        Address userRefundBtcAddress = PegTestUtils.createRandomBtcAddress(bridgeConstants.getBtcParams());
+        Address lpBtcAddress = PegTestUtils.createRandomBtcAddress(bridgeConstants.getBtcParams());
+
+        Repository repository = spy(createRepository());
+        // Set current bridge balance in order to create the condition to surpass the locking cap when needed
+        doReturn(co.rsk.core.Coin.valueOf(1)).when(repository).getBalance(any());
+
+        BtcBlockStoreWithCache btcBlockStore = mock(BtcBlockStoreWithCache.class);
+        BtcBlockStoreWithCache.Factory mockFactory = mock(BtcBlockStoreWithCache.Factory.class);
+        when(mockFactory.newInstance(repository, bridgeConstants, provider, activations)).thenReturn(btcBlockStore);
+        Block executionBlock = Mockito.mock(Block.class);
+        when(executionBlock.getNumber()).thenReturn(10L);
+
+        BtcLockSender btcLockSender = mock(BtcLockSender.class);
+        BtcLockSenderProvider btcLockSenderProvider = mock(BtcLockSenderProvider.class);
+        when(btcLockSenderProvider.tryGetBtcLockSender(any())).thenReturn(Optional.of(btcLockSender));
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withProvider(provider)
+            .withBridgeConstants(bridgeConstants)
+            .withBtcLockSenderProvider(btcLockSenderProvider)
+            .withActivations(activations)
+            .withBtcBlockStoreFactory(mockFactory)
+            .withExecutionBlock(executionBlock)
+            .withRepository(repository)
+            .build();
+
+        bridgeSupport = spy(bridgeSupport);
+        doReturn(lockingCap).when(bridgeSupport).getLockingCap();
+
+        Keccak256 derivationArgumentsHash = PegTestUtils.createHash3(0);
+        RskAddress lbcAddress = PegTestUtils.createRandomRskAddress();
+        Keccak256 fastBridgeDerivationHash = bridgeSupport.getFastBridgeDerivationHash(
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lpBtcAddress,
+            lbcAddress
+        );
+
+        Address activeFederationAddress = PegTestUtils.getFastBridgeAddressFromRedeemScript(
+            bridgeConstants,
+            activeFederation.getRedeemScript(),
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+        );
+
+        Address retiringFederationAddress = PegTestUtils.getFastBridgeAddressFromRedeemScript(
+            bridgeConstants,
+            retiringFederation.getRedeemScript(),
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+        );
+
+        BtcTransaction btcTx = btcTransactionProvider.provide(bridgeConstants, activeFederationAddress, retiringFederationAddress);
+        btcTx.addInput(
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes()),
```

Comment:
just a comment, the use of the fastBridgeDerivationHash here may lead to a mistake... It is not related and won't happen in reality. I would use a zero hash instead.
```suggestion
            Sha256Hash.ZERO_HASH,
```

Reply:
Done